src/commands/edge_app.rs

```diff
@@ -1,227 +1,777 @@
 use crate::authentication::Authentication;
-use crate::commands::CommandError;
+use crate::commands;
+use crate::commands::{
+    CommandError, EdgeAppManifest, EdgeAppSettings, EdgeAppVersions, EdgeApps, Setting,
+};
+use indicatif::{ProgressBar, ProgressStyle};
+use log::debug;
 use std::collections::HashMap;
+use std::{str, thread};
 
-use log::debug;
 use reqwest::header::HeaderMap;
 use reqwest::StatusCode;
 use serde::{Deserialize, Serialize};
+use serde_json::json;
 use serde_yaml;
 use std::fs;
 use std::fs::File;
 use std::io::Write;
 use std::path::Path;
+use std::time::Duration;
 
-#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
-pub struct EdgeAppManifest {
-    #[serde(default)]
-    pub id: String,
-    pub name: String,
-    pub version: String,
-    pub description: String,
-    pub icon: String,
-    pub author: String,
-    pub homepage_url: String,
-}
+use crate::commands::edge_app_utils::{
+    collect_paths_for_upload, detect_changed_files, detect_changed_settings,
+    ensure_edge_app_has_all_necessary_files, generate_file_tree, FileChanges, SettingChanges,
+};
 
 pub struct EdgeAppCommand {
     authentication: Authentication,
 }
 
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct AssetSignature {
+    pub(crate) signature: String,
+}
+
 impl EdgeAppCommand {
     pub fn new(authentication: Authentication) -> Self {
         Self { authentication }
     }
 
-    pub fn init(self, path: &Path) -> Result<(), CommandError> {
-        let mut object = serde_yaml::to_value(EdgeAppManifest::default())?;
-        let map = object.as_mapping_mut().ok_or(CommandError::MissingField)?;
+    pub fn create(&self, name: &str, path: &Path) -> Result<(), CommandError> {
+        let response = commands::post(
+            &self.authentication,
+            "v4/edge-apps?select=id,name",
```

Comment:
Hm - name will be the one you sent certainly.

Reply:
yeah - and easier to parse it from response then

Reply:
Fixed - merge issue